### PR TITLE
makefile: Make SELinux support configurable

### PR DESCRIPTION
**Pull request by andreabolognani**
SELinux support requires libselinux to be available, but that's not the only factor: for example, Fedora 31 has libselinux but not a version of selinux-policy that knows about Kata containers, so enabling SELinux support by default in that case causes usability issues.

Another issue with the current implementation is that, when libselinux is absent, SELinux support will be quietly disabled, which might not be what the user (or packager) intended.

To solve both problems, introduce the new `FEATURE_SELINUX` user variable. This variable takes one of three values:

  * check (default): keep the current behavior;

  * yes: enable SELinux support, erroring out if libselinux is
         not present on the system;

  * no: disable SELinux support.

In the future we might want to formalize support for optional build-time features, but for now this will do.

Fixes: #2623

### CODE REVIEW

**Comment by andreabolognani** [comment]
Updated to use make's `override` directive so that `make FEATURE_SELINUX=check` works properly.

The behavior with respect to build tags was already correct, but it erroneously reported

```
SELinux (FEATURE_SELINUX): check
```

instead of the detected value in the summary.

**Comment by fidencio** [comment]
- **jenkins-ci-podman-fedora-31:**
  - https://github.com/kata-containers/ci/issues/271

**Comment by fidencio** [comment]
**jenkins-ci-ubuntu-18-04-initrd:**
```
[2] export
[2] /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR-initrd/go/src/github.com/kata-containers/tests/integration/docker/export_test.go:15
[2]   export with docker
[2]   /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR-initrd/go/src/github.com/kata-containers/tests/integration/docker/export_test.go:32
[2]     export a container
[2]     /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR-initrd/go/src/github.com/kata-containers/tests/integration/docker/export_test.go:33
[2]       should export filesystem as a tar archive
[2]       /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR-initrd/go/src/github.com/kata-containers/tests/integration/docker/export_test.go:34
[2] ------------------------------
[2] load load with docker load a container 
[2]   should load image
[2]   /tmp/jenkins/workspace/kata-containers-runtime-ubuntu-18-04-PR-initrd/go/src/github.com/kata-containers/tests/integration/docker/load_test.go:38
[2] Running command '/usr/bin/docker [docker run --cidfile /tmp/cid136983232/qH8n1EB5Wb9QEuwCYyeFfW8zu3ihE2 --runtime kata-runtime -td --name qH8n1EB5Wb9QEuwCYyeFfW8zu3ihE2 busybox]'
[2] command failed error 'exit status 125'
[2] [docker run --cidfile /tmp/cid136983232/qH8n1EB5Wb9QEuwCYyeFfW8zu3ihE2 --runtime kata-runtime -td --name qH8n1EB5Wb9QEuwCYyeFfW8zu3ihE2 busybox]
[2] Timeout: 120 seconds
[2] Exit Code: 125
[2] Stdout: 87753f60b21198f0fadcd0128e5ae740fc6dc6ddd318c1abcca37485cc61ca1a
[2] 
[2] Stderr: docker: Error response from daemon: OCI runtime create failed: Failed to check if grpc server is working: rpc error: code = Unavailable desc = transport is closing: unknown.
[2] 
[2] Running command '/usr/bin/docker [docker rmi ]'
[2] command failed error 'exit status 1'
[2] [docker rmi ]
[2] Timeout: 120 seconds
[2] Exit Code: 1
[2] Stdout: 
[2] Stderr: Error: No such image: 
[2] 
[2] 
[2] • Failure in Spec Setup (BeforeEach) [4.004 seconds]
```

**Comment by andreabolognani** [comment]
@fidencio does the jenkins-ci-ubuntu-18-04-initrd error make any sense to you? I certainly can't see the connection to my changes...

**Comment by fidencio** [comment]
> @fidencio does the jenkins-ci-ubuntu-18-04-initrd error make any sense to you? I certainly can't see the connection to my changes...

It doesn't look related, at all. In any case, I'll re-run that specific CI again and let's see.

File: Makefile [hunk 1]
```diff
@@ -757,6 +776,9 @@ endif
 	@printf "\tKnown: $(sort $(HYPERVISORS))\n"
 	@printf "\tAvailable for this architecture: $(sort $(KNOWN_HYPERVISORS))\n"
 	@printf "\n"
+	@printf "• Features:\n"
+	@printf "\tSELinux (FEATURE_SELINUX): $(FEATURE_SELINUX)\n"
```
Comment (devimc) [inline Makefile:780]: maybe you want to replace one of these `FEATURE_SELINUX` with `SELINUXTAG` ... ?

Comment (andreabolognani) [reply Makefile:780]: `FEATURE_SELINUX` is the user-visible bit that can be tweaked at build time while `SELINUXTAG` is just an implementation detail, so I don't think it makes sense to display the latter.

**Pull request merged**